tests/Concerns/WithCustomQuerySizeTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Maatwebsite\Excel\Tests\Concerns;
+
+use Maatwebsite\Excel\Tests\TestCase;
+use Maatwebsite\Excel\Tests\Data\Stubs\Database\User;
+use Maatwebsite\Excel\Tests\Data\Stubs\Database\Group;
+use Maatwebsite\Excel\Tests\Data\Stubs\AfterQueueExportJob;
+use Maatwebsite\Excel\Tests\Data\Stubs\FromQueryWithCustomQuerySize;
+
+class WithCustomQuerySizeTest extends TestCase
+{
+    /**
+     * Setup the test environment.
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->loadLaravelMigrations(['--database' => 'testing']);
+        $this->loadMigrationsFrom(dirname(__DIR__) . '/Data/Stubs/Database/Migrations');
+        $this->withFactories(dirname(__DIR__) . '/Data/Stubs/Database/Factories');
+
+        factory(Group::class)->times(5)->create()->each(function ($group) {
+            $group->users()->attach(factory(User::class)->times(rand(1, 3))->create());
+        });
+
+        config()->set('excel.exports.chunk_size', 2);
+    }
+
+    /**
+     * @test
+     */
+    public function can_export_with_custom_count()
+    {
+        $export = new FromQueryWithCustomQuerySize();
+
+        $export->queue('export-from-query-with-count.xlsx', null, 'Xlsx')->chain([
+            new AfterQueueExportJob(dirname(__DIR__) . '/Data/Disks/Local/export-from-query-with-count.xlsx'),
+        ]);
+
+        $actual = $this->readAsArray(dirname(__DIR__) . '/Data/Disks/Local/export-from-query-with-count.xlsx', 'Xlsx');
+
+        $this->assertCount(Group::count(), $actual);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getPackageProviders($app)
+    {
+        if (class_exists(\Orchestra\Database\ConsoleServiceProvider::class)) {
```

Comment:
When would it be not available?

Reply:
Initially I thought to require orchestra/database conditionally, based on the build's env variables, but this didn't go as planned. So, now the console service provider will always be available.

I'll go ahead and remove the check.

Reply:
Done!